packages/solana-swap-provider/lib/SolanaSwapProvider.ts

```diff
@@ -0,0 +1,282 @@
+import { BigNumber, SwapParams, SwapProvider, Transaction } from '@liquality/types'
+import { Provider } from '@liquality/provider'
+import { Keypair, SystemProgram, PublicKey, TransactionInstruction } from '@solana/web3.js'
+import { deserialize } from 'borsh'
+import { base58, sha256 } from '@liquality/crypto'
+import { validateValue, validateSecretHash, validateExpiration } from '@liquality/utils'
+
+import {
+  Template,
+  initSchema,
+  claimSchema,
+  createInitBuffer,
+  createClaimBuffer,
+  createRefundBuffer,
+  InitData,
+  refundSchema
+} from './layouts'
+
+export default class SolanaSwapProvider extends Provider implements Partial<SwapProvider> {
+  doesBlockScan: boolean | (() => boolean)
+  signer: Keypair
+
+  generateSecret(message: string): Promise<string> {
+    return sha256(message)
+  }
+
+  async getSwapSecret(claimTxHash: string): Promise<string> {
+    const transactionByHash = await this.getMethod('getTransactionByHash')(claimTxHash)
+
+    const programId = transactionByHash._raw.programId.toString()
+
+    const initTxParams = await this.initTxParams(programId)
+
+    return initTxParams.secret_hash
+  }
+
+  async initiateSwap(swapParams: SwapParams, fee: number): Promise<Transaction<any>> {
+    const signer = await this.getMethod('getSigner')()
+
+    const programId = await this.getMethod('_deploy')(signer)
+
+    await this._waitForContractToBeExecutable(programId)
+
+    const { expiration, refundAddress, recipientAddress, value, secretHash } = swapParams
+
+    const initBuffer = createInitBuffer({
+      buyer: recipientAddress.toString(),
+      seller: refundAddress.toString(),
+      expiration,
+      secret_hash: secretHash,
```

Comment:
why `secret_hash` and not `secretHash` ?

Reply:
That goes directly to the rust model and over there camelCase is not the correct one instead of that snake_case is used